src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationService.kt

```diff
@@ -142,10 +146,10 @@ class ActivitiesMigrationService(
     }
 }
 
-private fun GetActivityResponse.toActivityMigrateRequest(): ActivityMigrateRequest = ActivityMigrateRequest(
+private fun GetActivityResponse.toActivityMigrateRequest(requestedStartDate: LocalDate): ActivityMigrateRequest = ActivityMigrateRequest(
   programServiceCode = programCode,
   prisonCode = prisonId,
-  startDate = startDate,
+  startDate = maxOf(startDate, requestedStartDate),
```

Comment:
If a migrated activity has a start date after the start date requested in the UI, we use that. This prevents future activities from starting before expected once they are migrated.